driver_test.go

```diff
@@ -12,26 +12,291 @@
+	if err != nil {
+		return nil, err
+	}
+
+	dataClient, err := spanner.NewClient(ctx, dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := &Connector{
+		ctx:         ctx,
+		client:      dataClient,
+		adminClient: adminClient,
+	}
+	return conn, nil
+}
+
+func CreateAdminClient(ctx context.Context) (*adminapi.DatabaseAdminClient, error) {
+
+	var adminClient *adminapi.DatabaseAdminClient
+	var err error
+
+	// Configure emulator if set.
+	if spannerHost, ok := os.LookupEnv("SPANNER_EMULATOR_HOST"); ok {
+		adminClient, err = adminapi.NewDatabaseAdminClient(
+			ctx,
+			option.WithoutAuthentication(),
+			option.WithEndpoint(spannerHost),
+			option.WithGRPCDialOption(grpc.WithInsecure()))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		adminClient, err = adminapi.NewDatabaseAdminClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return adminClient, nil
+}
+
+func (c *Connector) Close() {
+	c.client.Close()
+	c.adminClient.Close()
+}
+
+func init() {
+
+	var projectId, instanceId, databaseId string
+	var ok bool
+
+	// Get environment variables or set to default.
+	if instanceId, ok = os.LookupEnv("SPANNER_TEST_INSTANCE"); !ok {
+		instanceId = "test-instance"
+	}
+	if projectId, ok = os.LookupEnv("SPANNER_TEST_PROJECT"); !ok {
+		projectId = "test-project"
 	}
-	connector, err := driver.OpenConnector("projects/$PROJECT/instances/$INSTANCE/databases/$DATABASE")
+	if databaseId, ok = os.LookupEnv("SPANNER_TEST_DBID"); !ok {
+		databaseId = "gotest"
+	}
+
+	// Derive data source name.
+	dsn = "projects/" + projectId + "/instances/" + instanceId + "/databases/" + databaseId
+}
+
+// Executes DDL statements.
+func executeDdlApi(conn *Connector, ddls []string) error {
+
+	op, err := conn.adminClient.UpdateDatabaseDdl(conn.ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dsn,
+		Statements: ddls,
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(conn.ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Executes DML using the client library.
+func ExecuteDMLClientLib(dml []string) error {
+
+	// Open client/
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, dsn)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	db := sql.OpenDB(connector)
-	_ = db // Use db.
+	defer client.Close()
+
+	// Put strings into spanner.Statement structure.
+	var stmts []spanner.Statement
+	for _, line := range dml {
+		stmts = append(stmts, spanner.NewStatement(line))
+	}
+
+	// Execute statements.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		_, err := txn.BatchUpdate(ctx, stmts)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	return err
 }
+
+func TestQueryContext(t *testing.T) {
+
+	// Set up test table.
+	conn, err := NewConnector()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	err = executeDdlApi(conn, []string{`CREATE TABLE TestQueryContext (
+		A   STRING(1024),
+		B  STRING(1024),
```

Comment:
ultra-nit: alignment

Reply:
Changed to match the getting started guide